packages/react-components/src/components/VideoGallery/VideoGallery.tsx

```diff
@@ -105,6 +114,49 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
     return !!(layout === 'floatingLocalVideo' && remoteParticipants && remoteParticipants.length > 0);
   }, [layout, remoteParticipants]);
 
+  const containerRef = useRef<HTMLDivElement>(null);
+  const isMobileScreen = useIsSmallScreen(containerRef);
+  const visibleVideoParticipants = useRef<VideoGalleryParticipant[] | []>([]);
+  const visibleAudioParticipants = useRef<VideoGalleryParticipant[] | []>([]);
+  const [videoParticipants, setVideoParticipants] = useState<VideoGalleryParticipant[] | []>();
```

Comment:
this component should be pure and not have inbuilt state - what is this state for?